### PR TITLE
[SYCL] Reflect updates to get_pointer return type to relevant tests.

### PR DESCRIPTION
**Pull request by mmoadeli**
Reflect updates to get_pointer return type to relevant tests.
https://github.com/intel/llvm/pull/8493/

### CODE REVIEW

File: SYCL/Basic/group_async_copy.cpp [hunk 1]
```diff
@@ -122,22 +122,22 @@ template <typename T> int test(size_t Stride) {
        size_t Offset = GrId * WorkGroupSize;
        if (Stride == 1) { // Check the version without stride arg.
          auto E = NDId.async_work_group_copy(
-             Local.get_pointer(), In.get_pointer() + Offset, NElemsToCopy);
+             local_ptr<T>(Local), In.get_pointer() + Offset, NElemsToCopy);
```
Comment (aelovikov-intel) [inline SYCL/Basic/group_async_copy.cpp:125]: IIUC, this uses 

```
template <typename ElementType,
          access::decorated IsDecorated = access::decorated::legacy>
using local_ptr
```

deprecated value (`legacy`) for `access::decorated` template param. Is that possible to use either `yes` or `no` instead?

Comment (mmoadeli) [reply SYCL/Basic/group_async_copy.cpp:125]: @aelovikov-intel, Sounds reasonable, I would suggest doing so in another PR, though as it's not directly related to this one.

Comment (aelovikov-intel) [reply SYCL/Basic/group_async_copy.cpp:125]: Well, one can argue that such a change would be independent of your intel/llvm PR and consequently make this one redundant as well.

Also, the spec for `async_work_group_copy` says it only accepts ` decorated_local_ptr` so I'm not even sure what exactly happens here. Do we have an implicit conversion happening?

I guess I can give you an LGTM, but still...

**Review by aelovikov-intel — APPROVED** [review]

**Comment by mmoadeli** [comment]
@aelovikov-intel, thanks for the comments. An argument of **async_work_group_copy** is local_ptr<T> which is a multi_ptr. In multi_ptr class there is an implicit conversion from local_accessor to multi_ptr.
The updates made in the corresponding PR  changed the return type of get_pointer from local_ptr to T*.  The changes in this PR provides local_ptr which is needed by  **async_work_group_copy**.

**Comment by aelovikov-intel** [comment]
> An argument of **async_work_group_copy** is local_ptr

In the specification or in the implementation? Because per the spec at https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.members.nditem, all of the overloads have some kind of `decorated_*_ptr` which is a `multi_ptr` with `decorated::yes` template parameter while `local_ptr`'s default value is `legacy`.

**Pull request merged**